java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -347,6 +347,7 @@ public FlightInfo getCrossReference(String pkCatalog, String pkSchema, String pk
     private String token;
     private boolean useTls;
     private boolean disableCertificateVerification;
+    private boolean useSystemTrustStore;
```

Comment:
We should explicitly set defaults for disableCertificateVerification (false), useTls(true), and useSystemTrustStore(true) in the builder.

Reply:
I'm already doing this in the enum values inside `ArrowFlightConnectionConfigImpl`

```java
public enum ArrowFlightConnectionProperty implements ConnectionProperty {
    HOST("host", null, Type.STRING, true),
    PORT("port", null, Type.NUMBER, true),
    USER("user", null, Type.STRING, false),
    PASSWORD("password", null, Type.STRING, false),
    SSL("ssl", true, Type.BOOLEAN, false),
    CERTIFICATE_VERIFICATION("disableCertificateVerification", false, Type.BOOLEAN, false),
    TRUST_STORE("trustStore", null, Type.STRING, false),
    TRUST_STORE_PASSWORD("trustStorePassword", null, Type.STRING, false),
    USE_SYSTEM_TRUST_STORE("useSystemTrustStore", true, Type.BOOLEAN, false),
```

Reply:
Do we ever short-circuit using the connection property enum though and use the builder directly?

Reply:
No, we always use the enum. Only in tests, where we instantiate the ArrowFlightSQLHandler directly